app/src/main/java/app/nexd/android/ui/helper/detail/HelperDetailViewModel.kt

```diff
@@ -57,6 +66,18 @@ class HelperDetailViewModel(private val api: Api) : ViewModel() {
             .blockingSubscribe()
     }
 
+    private fun declineRequest(requestId: Long) {
+        val helpListIdFromRequest =
+            api.helpRequestsControllerGetSingleRequest(requestId).blockingFirst().helpListId
+        helpListIdFromRequest?.let { helpListId ->
+            api.helpListsControllerDeleteHelpRequestFromHelpList(helpListId, requestId)
+                .subscribeOn(io())
+                .doOnError {
+                    Log.e("Error", it.message.toString())
+                }.blockingSubscribe()
```

Comment:
Please dont use `blockingSubscribe` it is blocking the caller thread (in this case Main) and causes freezes. Also usage of doOnError is not a proper error handling. All thee doOn* operators are just side effects and do not catch the event/error. If an error would occur here, the app would crash. Please use a proper subscribe with onNext and onError handlers inside.

Reply:
fixed with recent push